File: edge-cases/java/wildcard-generics/Sample.java
package edge.generics;

import java.util.List;

public final class Sample {
    public double sum(List<? extends Number> values) {
        double total = 0.0;
        for (Number value : values) {
            total += value.doubleValue();
        }
        return total;
    }

    public void addOne(List<? super Integer> values) {
        values.add(1);
    }
}

File: fixtures/edge-static-j2k/wildcard-generics/Sample.kt
package edge.generics

class Sample {
    fun sum(values: List<out Number>): Double {
        var total = 0.0
        for (value in values) {
            total += value.toDouble()
        }
        return total
    }

    fun addOne(values: MutableList<in Int>) {
        values.add(1)
    }
}
